service/src/main/java/tds/session/repositories/SessionRepository.java

```diff
@@ -15,4 +15,18 @@
      * @return optional containing {@link tds.session.Session Session} otherwise empty Optional
      */
     Optional<Session> getSessionById(UUID id);
+
+    /**
+     * Pause an existing {@link Session}, updating the {@link Session}'s status to indicate it is no longer "open".
+     * <p>
+     *     The database design for microservices is to never execute an UPDATE against existing records.  Any changes
+     *     to the state of an object should be recorded as a new record.  For example, when a session is paused, its
+     *     state is changed.  Rather than updating the existing record in the session.session table, a new record should
+     *     be inserted into a newStatus table that records the change.
+     * </p>
+     *
+     * @param sessionId The id of the {@link Session} to pause
+     * @param newStatus A description of why the {@link Session} is being paused
+     */
+    void pause(final UUID sessionId, final String newStatus);
```

Comment:
Why are well allowing callers to pass in a `newStatus`?  Right now in the session table in deployment the only two values are `open` or `closed`.  It doesn't look to me that `session.status` column represents the reason but instead the actual status.